components/system-api/src/private.rs

```diff
@@ -269,57 +255,85 @@ pub struct ConsensusEnabledQuery {
 }
 
 /// Private system API.
-#[derive(Debug)]
+#[derive(Clone, Debug)]
 pub(super) struct SystemApi {
-    info: NodeInfo,
+    blockchain: Blockchain,
     shared_api_state: SharedNodeState,
     sender: ApiSender<ExternalMessage>,
+    start_time: SystemTime,
 }
 
 impl SystemApi {
     /// Create a new `private::SystemApi` instance.
-    pub fn new(sender: ApiSender<ExternalMessage>, shared_api_state: SharedNodeState) -> Self {
+    pub fn new(
+        blockchain: Blockchain,
+        sender: ApiSender<ExternalMessage>,
+        shared_api_state: SharedNodeState,
+    ) -> Self {
         Self {
+            blockchain,
             sender,
-            info: NodeInfo::new(),
             shared_api_state,
+            start_time: SystemTime::now(),
         }
     }
 
     /// Add private system API endpoints to the corresponding scope.
     pub fn wire(self, api_scope: &mut ApiScope) -> &mut ApiScope {
-        self.handle_peers_info("v1/peers", api_scope)
-            .handle_peer_add("v1/peers", api_scope)
-            .handle_network_info("v1/network", api_scope)
-            .handle_is_consensus_enabled("v1/consensus_enabled", api_scope)
-            .handle_set_consensus_enabled("v1/consensus_enabled", api_scope)
+        self.handle_info("v1/info", api_scope)
+            .handle_stats("v1/stats", api_scope)
+            .handle_peers("v1/peers", api_scope)
+            .handle_consensus_status("v1/consensus_status", api_scope)
             .handle_shutdown("v1/shutdown", api_scope);
         api_scope
     }
 
-    fn handle_peers_info(self, name: &'static str, api_scope: &mut ApiScope) -> Self {
+    fn handle_info(self, name: &'static str, api_scope: &mut ApiScope) -> Self {
+        let self_ = self.clone();
         let shared_api_state = self.shared_api_state.clone();
         api_scope.endpoint(name, move |_query: ()| -> api::Result<_> {
-            let mut outgoing_connections: HashMap<SocketAddr, OutgoingConnection> = HashMap::new();
+            let mut connected_peers = Vec::new();
 
             for connect_info in shared_api_state.outgoing_connections() {
-                outgoing_connections.insert(
-                    connect_info.address.parse().unwrap(),
-                    OutgoingConnection {
-                        public_key: Some(connect_info.public_key),
-                    },
-                );
+                connected_peers.push(ConnectedPeerInfo::new(&connect_info, "outgoing"));
+            }
+
+            for connect_info in shared_api_state.incoming_connections() {
+                connected_peers.push(ConnectedPeerInfo::new(&connect_info, "incoming"));
             }
 
-            Ok(PeersInfo {
-                incoming_connections: shared_api_state.incoming_connections(),
-                outgoing_connections,
+            Ok(NodeInfo {
+                consensus_status: self.get_consensus_status(),
```

Comment:
Nit: `get_consensus_status()` can be made a free function or a static method accepting `&SharedNodeState` as an argument. In this way, you wouldn't need to clone `self` above.